RDSC-4170: RDI 1.16.0 Release notes (#2443)
redis/docs: content/integrate/redis-data-integration/release-notes/rdi-1-16-0.md

  - Metric collection selection is now based on component type instead of name, avoiding misconfigured metrics.
- **API correctness and compatibility improvements**: Ensured backwards compatibility between API v1 and v2 and fixed issues where API v2 source/target dry-run endpoints could return HTTP 500 for certain invalid requests.
- **Environment variable handling for monitoring statistics**: Fixed an issue where `/api/v1/monitoring/statistics` could return HTTP 500 when the target database port was specified as an environment variable.

### Spanner integration with Flink

- **Enhanced Spanner disaster recovery with Redis Active-Active**: When using Cloud Spanner as a CDC source with a Redis Active-Active database, the Spanner Flink collector now handles long-running workloads and failover scenarios more reliably, improving the overall disaster recovery solution.
- **Spanner client resource management**: Spanner clients are now reused per database instead of being created repeatedly, eliminating a memory leak and reducing connection churn on Spanner.

> - **Spanner collector memory tuning**: Increased the default process and heap size for the Spanner Flink collector and aligned memory fractions with the main Flink processor so it can handle larger change streams without instability.

Reviewer: ```suggestion
- **Flink collector memory tuning**: Increased the default process and heap size for the Flink collector and tuned memory fractions for better performance and stability.
```
